### DISCUSSION TITLE
Custom FIM with external API

### DISCUSSION THREAD

**Discussion by gutbash** [rjmacarthy/twinny#388] [Q&A]
How do I use a provider like LiteLLM or something that is compatible with calling the OpenAI/Anthropic API to enable FIM non-locally? Has no one done this before?

**Comment by rjmacarthy** [comment]
Hey, some people have experimented with this See here: https://github.com/twinnydotdev/twinny/issues/142 It's possible to get something working already using custom templates and LiteLLM but when I tried the results were mixed, maybe with some more work and fiddling things could be improved.  Many thanks!
